src/pop/blockchain/mempool_block_tree.cpp

```diff
@@ -42,7 +42,7 @@ bool MemPoolBlockTree::checkContextually(const VbkBlock& block,
 
   auto* tip = vbkstable.getBestChain().tip();
   VBK_ASSERT(tip);
-  bool tooOld = abs(tip->getHeight() - block.getHeight()) >
+  bool tooOld = tip->getHeight() - block.getHeight() >
```

Comment:
Why?

Reply:
`tooOld` property is only for blocks which are below the tip

Reply:
I don't see the logic. Tip can be lower in case if it's POP score is higher than second chain, but second chain is higher.

Reply:
We are applying different blocks into the `MemPoolBlockTree` and it is possible that we will apply some blocks which are higher/next blocks to the current tip, but will not apply it into the tree.